TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Command/NodeCommandControllerPlugin.php

```diff
@@ -856,6 +861,95 @@ protected function removeNode($nodeIdentifier, $dimensionsHash)
+        } else {
+            $this->output->outputLine('Checking for nodes with invalid dimensions ...');
+            $nodesArray = [];
+            foreach ($this->nodeTypeManager->getNodeTypes() as $nodeType) {
+                /** @var NodeType $nodeType */
+                if ($nodeType->isAbstract()) {
+                    continue;
+                }
+                $nodesArray = array_merge($nodesArray, $this->collectNodesWithInvalidDimensions($nodeType, $workspaceName, $allowedDimensionCombinations));
+            }
+        }
+
+        if (count($nodesArray) === 0) {
+            return;
+        }
+
+        if (!$dryRun) {
+            $self = $this;
+            $this->output->outputLine();
+            $this->output->outputLine('Nodes with invalid dimension values found.' . PHP_EOL . 'You might solve this by migrating them to your current dimension configuration or by removing them.');
+            $this->askBeforeExecutingTask(sprintf('Do you want to remove %s node%s with invalid dimensions now?', count($nodesArray), count($nodesArray) > 1 ? 's' : ''), function () use ($self, $nodesArray, $workspaceName) {
+                foreach ($nodesArray as $nodeArray) {
+                    $self->removeNode($nodeArray['identifier'], $nodeArray['dimensionsHash']);
+                }
+                $self->output->outputLine('Removed %s node%s with invalid dimension values.', array(count($nodesArray), count($nodesArray) > 1 ? 's' : ''));
+            });
+        } else {
+            $this->output->outputLine('Found %s node%s with invalid dimension values to be removed.', array(count($nodesArray), count($nodesArray) > 1 ? 's' : ''));
+        }
+        $this->output->outputLine();
+    }
+
+    /**
+     * Collects all nodes of the given node type which have dimension values not fitting to the current dimension
+     * configuration.
+     *
+     * @param NodeType $nodeType
+     * @param string $workspaceName
+     * @param array $allowedDimensionCombinations
+     * @return array
+     */
+    protected function collectNodesWithInvalidDimensions(NodeType $nodeType, $workspaceName, array $allowedDimensionCombinations)
+    {
+        $nodes = [];
+
+        /** @var QueryBuilder $queryBuilder */
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+
+        $queryBuilder->select('n')
+            ->from('TYPO3\TYPO3CR\Domain\Model\NodeData', 'n')
+            ->where('n.nodeType = :nodeType')
+            ->andWhere('n.workspace = :workspace')
+            ->andWhere('n.movedTo IS NULL OR n.removed = :removed')
+            ->setParameter('nodeType', $nodeType)
+            ->setParameter('workspace', $workspaceName)
+            ->setParameter('removed', false, \PDO::PARAM_BOOL);
+
+        foreach ($queryBuilder->getQuery()->getArrayResult() as $nodeDataArray) {
+            if ($nodeDataArray['path'] === '/') {
```

Comment:
Mmmm, what about `/sites` ? And in the future maybe other node trees that do not use the dimensions?
Sounds dangerous ;)